Mathlib/LinearAlgebra/RootSystem/Basic.lean

```diff
@@ -156,15 +169,23 @@ lemma coreflection_apply (f : N) :
     P.coreflection i f = f - (P.toLin (P.root i) f) • P.coroot i :=
   rfl
 
-lemma bijOn_reflection_root :
-    BijOn (P.reflection i) (range P.root) (range P.root) :=
-  ((finite_range P.root).injOn_iff_bijOn_of_mapsTo (P.mapsTo_preReflection_root i)).mp <|
-    (P.reflection i).injective.injOn _
+  @[simp]
+lemma coreflection_apply_self :
+    P.coreflection i (P.coroot i) = - P.coroot i :=
+  Module.reflection_apply_self (P.flip.coroot_root_two i)
+
+lemma coreflection_eq_flip_reflection (f : N) : P.coreflection i f = P.flip.reflection i f :=
+  rfl
+
+@[simp]
+lemma coreflection_same (x : N) : P.coreflection i (P.coreflection i x) = x :=
```

Comment:
We generally use `self` when a variable `x` occurs in two places, and `of_same` when there is `i`, `j` and an assumption `h : i = j`.
```suggestion
lemma coreflection_self (x : N) : P.coreflection i (P.coreflection i x) = x :=
```